core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -2111,7 +2111,6 @@ protected void copyToInbound(DefaultMuleMessage newMessage) throws Exception
         }
 
         newMessage.clearProperties(PropertyScope.INBOUND);
-        newMessage.clearProperties(PropertyScope.INVOCATION);
```

Comment:
I thought this was wrong, but now see it's used by VM transport. I'll revert this and create response message differently in TestMessageDispatcher.